operations/security_and_authentication/Update_default_Air-Cooled_BMC_and_Leaf_switch_SNMP_credentials.md

```diff
@@ -0,0 +1,196 @@
+    {
+        "SNMPUsername": "testuser",
+        "SNMPAuthPassword": "foo",
+        "SNMPPrivPassword": "bar"
+    }
+    ```
+
+2.  Update the default credentials in `customizations.yaml` for REDS and HMS Discovery to work with:
+    
+    Specify the desired default Redfish credentials:
+    ```bash
+    ncn-m001# echo '{"Cray":{"Username":"root","Password":"foobar"}}' | base64 > reds.redfish.creds.json.b64
+    ```
+
+    Specify the desired default SNMP credentials:
+    ```bash
+    ncn-m001# echo '{"SNMPUsername":"testuser","SNMPAuthPassword":"foo1","SNMPPrivPassword":"bar2"}' | base64 > reds.switch.creds.json.b64
+    ```
+
+
+    Update and regenerate `cray_reds_credentials` sealed secret:
+    ```bash
+    ncn-m001# cat << EOF | yq w - 'data.vault_redfish_defaults' "$(<reds.redfish.creds.json.b64)" | yq w - 'data.vault_switch_defaults' "$(<reds.switch.creds.json.b64)" | yq r -j - | ./utils/secrets-encrypt.sh | yq w -f - -i ./customizations.yaml 'spec.kubernetes.sealed_secrets.cray_reds_credentials'
+    {
+        "kind": "Secret",
+        "apiVersion": "v1",
+        "metadata": {
+            "name": "cray-reds-credentials",
+            "namespace": "services",
+            "creationTimestamp": null
+        },
+        "data": {}
+    }
+    EOF
+    ```
+
+4.  Decrypt generated secret for review.
+    
+    Default Redfish credentials:
+    ```bash
+    ncn-m001# ./utils/secrets-decrypt.sh cray_reds_credentials ./certs/sealed_secrets.key ./customizations.yaml | jq .data.vault_redfish_defaults -r | base64 -d | jq
+    {
+        "Username": "root",
+        "Password": "foobar"
+    }
+    ```
+
+    Default Switch SNMP credentials:
+    ```bash
+    ncn-m001# ./utils/secrets-decrypt.sh cray_reds_credentials ./certs/sealed_secrets.key ./customizations.yaml | jq .data.vault_switch_defaults -r | base64 -d | jq
+    {
+        "SNMPUsername": "testuser",
+        "SNMPAuthPassword": "foo1",
+        "SNMPPrivPassword": "bar2"
+    }
+    ```
+
+
+5.  Update the site-init secret for the system:
+    ```bash
+    ncn-m001# kubectl delete secret -n loftsman site-init
+    ncn-m001# kubectl create secret -n loftsman generic site-init --from-file=customizations.yaml
+    ```
+
+#### 1.3 Redeploy REDS to pick up the new sealed secret and push credentials into vault.
+1.  Determine the version of REDS:
+    ```bash
+    ncn-m001# REDS_VERSION=$(kubectl -n loftsman get cm loftsman-core-services -o jsonpath='{.data.manifest\.yaml}' | yq r - 'spec.charts.(name==cray-hms-reds).version')
+    ncn-m001# echo $REDS_VERSION
+    ```
+
+2.  Create `reds-manifest.yaml`:
+    ```bash
+    ncn-m001# cat > reds-manifest.yaml << EOF 
+    apiVersion: manifests/v1beta1
+    metadata:
+        name: reds
+    spec:
+        charts:
+        - name: cray-hms-reds
+          version: $REDS_VERSION
+          namespace: services
+    EOF
+    ```
+
+2.  Merge customizations.yaml with reds-manifest.yaml
```

Comment:
Missing punctuation at the end of this step

Reply:
These file names are surrounded with backticks in text/steps above. This should be consistent.

Reply:
Should be resolved fixed now.